### ISSUE TITLE
Allow selection during installation using pacman when multiple providers are available

### ISSUE THREAD

**Issue by necrosisy** [archlinux/archinstall#2188]
Some software packages may have multiple providers as dependencies. During the automatic installation process, archinstall selects the first project in lexicographic order by default. Perhaps it would be more appropriate to add an option to manually operate when providers are selected during the installation process, just like the normal pacman installation process.

**Comment by Torxed** [comment]
I think I understand what you're pointing towards, but could you give a concrete example of a package that would trigger this?

**Comment by necrosisy** [comment]
> 我想我明白你的意思，但是你能给出一个会触发这个问题的包的具体例子吗？

Of course, for example, when using the kde desktop configuration file, the dependency phonon-qt5-backend has multiple providers. And when the gstreamer and vlc backends are installed at the same time, kde uses gstream by default, and I have to uninstall it manually. switch. Overall I think as a feature this can help make archinstall easier to use.

**Comment by necrosisy** [comment]
> I think I understand what you're pointing towards, but could you give a concrete example of a package that would trigger this?

Of course, for example, when using the kde desktop configuration file, the dependency phonon-qt5-backend has multiple providers. And when the gstreamer and vlc backends are installed at the same time, kde uses gstream by default, and I have to uninstall it manually switch. Overall I think as a feature this can help make archinstall easier to use.

**Comment by Torxed** [comment]
Thank you, I have no quick solution to this as it will require some form of halt-and-ask logic that we currently don't have.

We're trying to ask all the user input up-front, so halting to ask questions doesn't really appeal to the workflow that I've built over the years. So I'd need to investigate if it's possible via `libalpm` or something ahead of time to iterate over the providers - as this would be the ideal way to do it.

**Comment by svartkanin** [comment]
I think this may start spinning too much. If people choose a full automated installation then it would get interrupted potentially. Is that wanted? Not wanted?
What's the downside of us using a fixed provider?
